oauth2_client/client.py

```diff
@@ -43,63 +101,68 @@ def request(self, method, url, *args, **kwargs):  # pylint: disable=arguments-di
 
         Returns:
             Response object.
+
+        Raises:
+            TokenExpiredError: 1) when token expiry detected and `expired_callback` not specified;
+                2) when 403 Forbidden status code received and `expired_callback` not specified;
+                3) when `expired_callback` is specified and more than one token re-fetch happens
+                within `REAUTH_LIMIT_SECONDS` period.
         """
         absolute_url = urljoin(self.service_host, url)
-        return super(OAuth2Client, self).request(method, absolute_url, *args, **kwargs)
-
-
-def get_client(service_name):
+        try:
+            # super().request may raise `TokenExpiredError`
+            resp = super(OAuth2Client, self).request(method, absolute_url, *args, **kwargs)
+            if resp.status_code == 403:
```

Comment:
Doublechecking. Could 403 error to come due to another source other than `access token is expired`, or it is only raised when the token is expired?

Reply:
And if you have status code, why you need exception?

Reply:
resolved in a call